### PR TITLE
Fixing keyboard up-arrow navigation and keyboard accelerators on MenuStrip

### LINKED ISSUE #1021: Keyboard accelerators on MenuStrip are ignored
* .NET Core Version: Preview6 27720-09
* Have you experienced this same bug with .NET Framework?: No

**Problem description:**

Keyboard accelerators on MenuStrip are ignored. For instance, when the File menu is open and you press 'S' to save, nothing happens, despite the menu item text being "&Save". Also, the keyboard arrows don't work fully - when you press up-arrow on the first menu item, it doesn't cycle to the last item on the menu as it does in .NET Framework (and every other Windows app). This makes it hard work to navigate to the last item on the menu.

(Note that this bug was likely introduced around Preview5. In earlier .NET Core previews, I recall menu navigation working correctly.)

**Minimal repro:**

```
var file = new ToolStripMenuItem ("File");

file.DropDownItems.Add ("&New");
file.DropDownItems.Add ("&Open");
file.DropDownItems.Add ("&Save");
file.DropDownItems.Add ("Save &as");

var main = new MenuStrip ();
main.Items.Add (file);

var form = new Form { MainMenuStrip = main };
form.Controls.Add (main);

form.Show();
```

### PR DESCRIPTION
**Pull request by vladimir-krestov**
Fixes #1021

## Proposed changes
- Fix `CanProcessMnemonic` method implementation to correct work of keyboard accelerators
- Change the implementation of getting the `start` item in `GetNextItemHorizontal` method to correct work of up\down cycle navigation 

### Before
- Issue 1: keyboard accelerators on MenuStrip are ignored
- Issue 2: keyboard up-arrow return the first item when first item is focused. Focus does not move cyclically to the last element.

### After
- Keyboard accelerators of MenuItems work well in MenuStrip/ToolStrip/ContextMenuStrip/StatusStrip controls
- Keyboard up-arrow return the last item when first item is focused. Keyboard up\down cycle navigation works correctly

## Test methodology 
- Manual keyboard testing

## Test environment(s) 
- Microsoft Windows [Version 10.0.18362.175]
- .NET Core SDK. Version: 3.0.0-preview8-27916-13

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 1]
```diff
@@ -2515,7 +2516,8 @@ private ToolStripItem GetNextItemHorizontal(ToolStripItem start, bool forward)
                 {  // provide negative wrap if necessary
                     current = (--current < 0) ? count + current : current;
                 }
-                if (this is ToolStripDropDown dropDown)
+
+                if (dropDown != null)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs]: Merge with the nested `if`?

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 2]
```diff
@@ -2530,9 +2532,26 @@ private ToolStripItem GetNextItemHorizontal(ToolStripItem start, bool forward)
...
             return null;
         }
 
+        private ToolStripItem GetStartItem(bool forward, bool isDropDown)
+        {
+            if (forward)
+            {
+                return DisplayedItems[DisplayedItems.Count - 1];
+            }
+            else if (!isDropDown)
```
[the remark this answers is not in the document]
Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs]: Thanks, fixed.

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs (hunk 1 above)
Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs]: Thanks, I fixed the nested `if`.

**Comment by vladimir-krestov** [comment]
@RussKie I added the unit tests to test keyboard arrow navigation and keyboard accelerators on ToolStrip. Made code refactoring.

**Comment by RussKie** [comment]
Thank you

**Pull request merged**